index.js

```diff
@@ -78,11 +81,16 @@ function handleError(error) {
 }
 
 if (process.argv.length > 2) {
-  process.argv.slice(2).forEach(function(arg, index) {
-    readAuth(function(err, auth) {
-      if(err) handleError(err);
+  readAuth(function(err, auth) {
+    if(err) handleError(err);
+    process.argv.slice(2).forEach(function(arg, index) {
       readExpenseFile(arg, function(expenses) {
-        sendExpenses(auth, expenses);
+        console.log("Sending " + expenses.length + " expenses:");
```

Comment:
Just a pointer: while refactoring, remember that logging is another responsibility.
Try to isolate/detach it from the implementations. What if instead of just printing to the console we wanted to save into a file or db? Should other modules know about logging or should we use it only when using the combining them?